kale/embed/hgt.py

```diff
@@ -0,0 +1,169 @@
+# Created by Peizhen Bai from modifying https://github.com/acbull/pyHGT
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torch_geometric.nn.conv import MessagePassing
+from torch_geometric.nn.inits import glorot
+from torch_geometric.utils import softmax
+import math
+
+
+class HGTConv(MessagePassing):
+    """
+    Heterogeneous Graph Transformer (HGT) is a GNN architecture for dedicated heterogeneous graph. The model is from
+    the '"Heterogeneous Graph Transformer "<https://arxiv.org/abs/2003.01332>'_ (WWW 2020) paper. The code is based on
+    'pyHGT'<https://github.com/acbull/pyHGT>_ source repo with slight modification.
+    Args:
+        in_dim: input embedding dimensions (required).
+        out_dim: output embedding dimensions (required).
+        num_types: the number of node types (required).
+        num_relations: the number of relation types (required).
+        n_heads: the number of attention heads (required).
+        dropout: dropout rate (default=0.2)
+        use_norm: whether to use normalization (default=True).
+        use_rte: whether to use relative temporal encoding (default=True).
```

Comment:
rte is quite unreadable, suggestions: relative_temporal or rel_temporal or rel_temp